test/index.js

```diff
@@ -1154,6 +1168,51 @@ describe('restify-client tests', function () {
         });
     });
 
+    it('GH-146: JSONClient should errback on receiving invalid JSON',
+    function (done) {
+        JSON_CLIENT.get('/json/invalid', function (err, req, res, data) {
+            assert.ok(err);
+            assert.deepEqual(err.name, 'RestError');
+            assert.deepEqual(err.message, 'Invalid JSON in response');
+            assert.deepEqual(err.cause().name, 'SyntaxError');
+            assert.include(err.cause().message, 'Unexpected end of');
+            assert.equal(200, res.statusCode);
```

Comment:
Could you add an assert for `data`? To be sure that it's `undefined` and not an empty object.